### ISSUE TITLE
IS it Bug?? Mongoengine seems to be unable to find db_field

### ISSUE THREAD

**Issue by ghost** [hmarr/mongoengine#512]
If you try this code.. you can see the problem I have..

```
class Embedded(EmbeddedDocument):
    boxfluxInt = IntField(default=0, db_field='i')
    meta = {'allow_inheritance': False}
class Test(Document):
    boxflux = MapField(field=EmbeddedDocumentField(Embedded), db_field='x')
    meta = {'collection': 'test',
                   'allow_inheritance': False}
Test.drop_collection()

newTestDoc = Test()
newTestDoc.boxflux['DICTIONARY_KEY'] = Embedded(boxfluxInt=1)
newTestDoc.save()
Test.objects.update_one(inc__boxflux__DICTIONARY_KEY__boxfluxInt=1)
```

The result in Mongodb is like..

```
> db.test.findOne()
{
    "_id" : ObjectId("4fbdbbc8c450190a50000001"),
    "x" : {
            "DICTIONARY_KEY" : {
                    "boxfluxInt" : 1,
                    "i" : 1
            }
    }
}
>
```

As you can see, I intended to increase 'x.DICTIONARY_KEY.i' by 1
but the result is that a new key (boxfluxInt) is created even though I set 'boxfluxInt' 's db_field as 'i'

Is it bug? or am I wrong?

I think the dictionary key ('DICTIONARY_KEY') makes conversion to mongo style db fields impossible.. if I'm correct..

**Comment by rozza** [comment]
Hi,

Yes that is a bug - have fixed will be in 0.6.11 out today.  Thanks for the bug report.

**Comment by ghost** [comment]
WOW!!! COOL!!
Rozza you are genius!!
